--- .orig/boot/sbin/init.shr.bootr
+++ /boot/sbin/init.shr.bootr
@@ -45,13 +45,38 @@
 mknod /dev/mmcblk0p2 b 179 2
 mknod /dev/mmcblk0p3 b 179 3
 
+mknod /dev/loop0 b 7 0
+mknod /dev/loop1 b 7 1
+mknod /dev/loop2 b 7 2
+mknod /dev/loop3 b 7 3
+mknod /dev/loop4 b 7 4
+mknod /dev/loop5 b 7 5
+mknod /dev/loop6 b 7 6
+mknod /dev/loop7 b 7 7
+
+sh /bootr/bin/set-bootr.sh
+
 lvm.static vgscan --ignorelockingfailure
-lvm.static lvchange -ay --ignorelockingfailure /dev/mapper/store-root
+lvm.static lvs /dev/store/fso >/dev/null 2>/dev/null || false
+if [ $? -eq 0 ]; then
+    lvm.static lvchange -ay --ignorelockingfailure /dev/mapper/store-fso
 
-# May need to say "don't update mtab" option below
-mount -o ro /dev/mapper/store-root /realroot
+    # May need to say "don't update mtab" option below
+    mount -o rw /dev/mapper/store-fso /realroot
 
-/sbin/pivot_root /realroot /realroot/boot
+    /sbin/pivot_root /realroot /realroot/boot
+else
+    lvm.static lvchange -ay --ignorelockingfailure /dev/mapper/store-media
+
+    # May need to say "don't update mtab" option below
+    mount -t vfat -o rw /dev/mapper/store-media /realroot
+
+    # mount boot.ext3 from 'usb partition'/android
+    mount -o loop /realroot/shr/rootfs.ext3 /realroot/shr/mnt
+
+    /sbin/pivot_root /realroot/shr/mnt /realroot/shr/mnt/boot
+    #mount --move /boot/realroot /mnt/sdcard
+fi
 
 # Recommended:
 
