--- .orig/boot/sbin/init.shr.bootr
+++ /boot/sbin/init.shr.bootr
@@ -45,11 +45,13 @@
 mknod /dev/mmcblk0p2 b 179 2
 mknod /dev/mmcblk0p3 b 179 3
 
+sh /bootr/bin/set-bootr.sh
+
 lvm.static vgscan --ignorelockingfailure
-lvm.static lvchange -ay --ignorelockingfailure /dev/mapper/store-root
+lvm.static lvchange -ay --ignorelockingfailure /dev/mapper/store-fso
 
 # May need to say "don't update mtab" option below
-mount -o ro /dev/mapper/store-root /realroot
+mount -o rw /dev/mapper/store-fso /realroot
 
 /sbin/pivot_root /realroot /realroot/boot
 
